rpc/interface/src/rpc_service.rs

```diff
@@ -32,7 +32,7 @@ pub trait RpcService<Request, Response>:
         Request,
         Response = Response,
         Error = RpcError,
-        Future: Future<Output = Result<Response, RpcError>> + Send + Sync + 'static,
+        Future: Future<Output = Result<Response, RpcError>> + Send + 'static,
```

Comment:
https://github.com/Cuprate/cuprate/pull/266#discussion_r1745799658 and also `BoxFuture` isn't `Sync` so we have to remove this.